### DISCUSSION TITLE
Language structure of scr files

### DISCUSSION THREAD

**Discussion by shawnthompson** [gc-da11yn/gc-da11yn.github.io#56] [Development (Open discussions)]
@duboisp @gcharest @calvinrodo

I followed this post:

[Multilingual sites with Eleventy](https://multilingual.lea-tortay.com/en/blog/first-post/)

I'm using the same file name for both languages, it was easier for me to set using the locale variable.

https://github.com/gc-da11yn/gc-da11yn.github.io/blob/main/src/_includes/partials/header.njk#L1

```
{% set otherLang = "fr" if locale == "en" else "en" %}
{% set otherLangugae = "Français" if locale == "en" else "English" %}
{% set toggleURL = page.filePathStem | replace("/en/", "fr/") if locale == "en"  else page.filePathStem | replace("/fr/", "en/") %}
```

Which is used:

```
{% if needsTranslation != true %}
<section id="wb-lng" class="col-xs-3 col-sm-12 pull-right text-right">
	<h2 class="wb-inv">{{ header[locale].languageSelections }}</h2>
	<ul class="list-inline mrgn-bttm-0">
		<li>
			<a lang="{{ otherLang }}" hreflang="{{ otherLang }}" href="{{ rootPath }}{{ toggleURL | replace("index", "") }}">
				<span class="hidden-xs">{{ otherLangugae }}</span>
				<abbr title="{{ otherLangugae }}" class="visible-xs h3 mrgn-tp-sm mrgn-bttm-0 text-uppercase">{{ otherLang }}</abbr>
			</a>
		</li>
	</ul>
</section>
{% endif %}
```

We could

**Comment by gcharest** [comment]
I'm new to eleventy so have to familiarize with posts/pages metadata ([Front matter in Jekyll](https://jekyllrb.com/docs/front-matter/) if you will). Using the file names only in a folder structure could work out for testing purposes but I did appreciate having the ref tag in Jekyll, linking two pages with the same ref regardless of the filename.

The other thing I noticed is we currently have the en/fr toggle but the rest of the URL isn't bilingual itself. (Not sure if we feel like tackling in a separate issue?)

**Comment by shawnthompson** [comment]
Having used both Jekyll and 11ty, you can pretty much do all the same things except instead of using Ruby Gems, I would think you would use NodeJS packages.

I wonder if we can use one source file and add both languages in the source, but have 11ty create two separate files in the `_site`.

**Reply by shawnthompson** [reply]
Adding a11yproject issue:

https://github.com/a11yproject/a11yproject.com/issues/511

**Reply by gcharest** [reply]
I'd probably try to keep it simple. Separation of files at this point works so might just focus on making it easy for starters. Could work on the side to help solve larger issues around i18n that others seem to be experiencing.

**Reply by duboisp** [reply]
> but have 11ty create two separate files in the _site

I think that will confuse more the web author when they will edit the page. Keeping language separately is easier to edit content page. 
But I can see beneficial if all those translated content file share one unique file with the meta data information to be reused on each page.

**Reply by shawnthompson** [reply]
So let's keep it as is for now and we can always play around later with it.
